addons/mail/models/mail_message.py

```diff
@@ -737,7 +737,9 @@ def unlink(self):
         messages_by_partner = defaultdict(lambda: self.env['mail.message'])
         partners_with_user = self.partner_ids.filtered('user_ids')
         for elem in self:
-            for partner in elem.partner_ids & partners_with_user:
+            for partner in (
```

Comment:
Actually I would remove all that code: removing records is something you globally never do (you archive business records), unless you did mistakes. I would avoid adding multiple queries and bus notifications, just gracefully handle any stuff linked to records that do not exists in any way.

Reply:
Would say it is easier to handle corner cases by dismissing stuff that do not do anything anymore, and keep model code simple. We should improve and cleanup standard use cases, and let corner cases flow without crashes, instead of trying to be perfect and adding code complexity, even if that means some incorrect numbers sometimes (even discord does not try to be consistent, and they have way more knowledge than us in that matter).